### DISCUSSION TITLE
building WRF HYDRO with NOAH NWM

### DISCUSSION THREAD

**Discussion by jbensabat** [NCAR/wrf_hydro_nwm_public#715] [Q&A]
Hello 
I am trying to build WRF HYDRO with NOAH and parallel option 
for this there is a (this is my understanding) to define **MPP_LAND**  during the building process
looking at the code of the driver Noah_hrldas_driver.f I see another preprocessor variable **_PARALLEL_**
so here are the questions

do I need to set both ?
if not what is the difference between them ?
which one should I set
best
jac

**Comment by scrasmussen** [comment]
Hi @jbensabat ,

Are you specifically trying to build with Noah for some reason? Because for a stand-alone model you would want to build with NoahMP. The easiest and quickest way to do this is to look at the [BUILD.md](https://github.com/NCAR/wrf_hydro_nwm_public/blob/main/docs/BUILD.md) file and follow the [CMake instructions](https://github.com/NCAR/wrf_hydro_nwm_public/blob/main/docs/BUILD.md#cmake-build). 

The default build is a parallel build which is as easy to do as:
```
$ mkdir build
$ cd build
$ cmake ..
$ make -j 4
```

If you have any other questions just let us know!

**Comment by jbensabat** [comment]
Good morning and thanks for your message.
I am learning the code and I think (at least thought) that using NWM first could be a good primer for the understanding of the results (simpler LSM)
I built the NWMMP version and could run it.
I have built the NWM version but there seem to be a number of of problems. one of them I am dealing with is in routine
subroutine output_chrt_NWM(domainId) file module_NWM_IO.f

   if(myId .eq. 0) then
      if(nlst(domainId)%channel_option .ne. 3) then
         ! Read in index values from Routelink that will be used to sort output
         ! variables by ascending feature_id.

         **iret = orchestrator%IO_Manager%netcdf_layer%open_file(trim(nlst(1)%route_link_f),NF90_NOWRITE,ncid=ftnRt)**
**in this call netcdf_layer is a NULL pointer causing the program to crash**

         !iret = nf90_open(trim(nlst(1)%route_link_f),NF90_NOWRITE,ncid=ftnRt)
         call nwmCheck(diagFlag,iret,'ERROR: Unable to open RouteLink file for index extraction')
         iret = nf90_inq_varid(ftnRt,'ascendingIndex',indVarId)
         if(iret .ne. 0) then
            call postDiagMsg(diagFlag,'WARNING: ascendingIndex not found in RouteLink file. No resorting will take place.')
            ascFlag = 0
         endif
         if(ascFlag .eq. 1) then
            iret = nf90_get_var(ftnRt,indVarId,chIndArray)
            call nwmCheck(diagFlag,iret,'ERROR: Unable to extract ascendingIndex from RouteLink file.')
         endif
         iret = nf90_close(ftnRt)
         call nwmCheck(diagFlag,iret,'ERROR: Unable to close RouteLink file.')
      else
         ascFlag = 0
      endif

I have observed that there are differences between version 5.2 from the tarball and the one in the GITHUB
this is the rationale for my question 
best
jac
